lexos/models/topword_model.py

```diff
@@ -0,0 +1,359 @@
+        for index, word in enumerate(words):
+            p_1 = count_list_i[index] / i_sum
+            p_t = count_list_j[index] / j_sum
+            z_score = TopwordModel._z_test(p1=p_1, pt=p_t, n1=i_sum, nt=j_sum)
+            # Record the significant result. (See details: _z_test())
+            if abs(z_score) >= 1.96:
+                word_score_dict.update({word: z_score})
+
+        # Sort word score dict by z-score in descending order.
+        sorted_dict = OrderedDict(sorted(word_score_dict.items(),
+                                         key=lambda item: abs(item[1]),
+                                         reverse=True))
+        # Convert the sorted result to a panda series.
+        result_series = pd.Series(sorted_dict)
+        return result_series
+
+    def _analyze_fill_to_all(self) -> ReadableResult:
+        """Detect if a given word is an anomaly.
+
+        While doing so, this method compares the occurrence of a given word
+        in a particular segment to the occurrence of the same word in the whole
+        corpus.
+        :return: a list of series, each series has a readable name and the
+                 result, which contains words with corresponding z-scores.
+        """
+        # Trap possible empty input error.
+        assert not self._doc_term_matrix.empty, SEG_NON_POSITIVE_MESSAGE
+
+        # Initialize, get all the file labels.
+        labels = [self._id_temp_label_map[file_id]
+                  for file_id in self._doc_term_matrix.index.values]
+        # Get word count in the whole corpus of each word.
+        word_count_sum = np.sum(self._doc_term_matrix.values, axis=0)
+
+        # Generate analysis result.
+        result_list = [TopwordModel._z_test_word_list(
+            count_list_i=row,
+            count_list_j=word_count_sum,
+            words=self._doc_term_matrix.columns.values)
+            for row in self._doc_term_matrix.values]
+
+        # Attach readable name to each result series.
+        readable_result_list = [
+            result_list[index].rename('Document "' + label +
+                                      '" compared to the whole corpus')
+            for index, label in enumerate(labels)]
+
+        return readable_result_list
+
+    def _analyze_class_to_all(self, division_map: pd.DataFrame) -> \
+            ReadableResult:
+        """Detect if a given word is an anomaly.
+
+        While doing so, this method compares the occurrence of a given word
+        in a particular segment to the occurrence of the same word in other
+        class of segments.
+        :return: a list of series, each series has a readable name and the
+                 result, which contains words with corresponding z-scores.
+        """
+        # Trap possible empty input error.
+        assert not self._doc_term_matrix.empty, SEG_NON_POSITIVE_MESSAGE
+
+        # Initialize all the labels and result to return.
+        readable_result = []
```

Comment:
Let me guess, you are mutating variable again? 
I understand my old code is much worse than this, but since you are converting it, you probably want to change all of it.

Reply:
Sorry, I knew this was bad, but I couldn't find a good way to solve it... I'll think about it more and try to fix other errors as well... So I'll close this pull request for now... 
Thanks a lot and all the advice are greatly appreciated.